esrally/driver/driver.py

```diff
@@ -710,10 +736,21 @@ def start_benchmark(self):
                     worker = self.target.create_client(host, self.config)
 
                     client_allocations = ClientAllocations()
+                    worker_client_contexts = {}
                     for client_id in clients:
                         client_allocations.add(client_id, self.allocations[client_id])
                         self.clients_per_worker[client_id] = worker_id
-                    self.target.start_worker(worker, worker_id, self.config, self.track, client_allocations)
+                        client_context = ClientContext(client_id=client_id, parent_worker_id=worker_id)
+
+                        if create_api_keys:
+                            resp = self.create_api_key(self.default_sync_es_client, client_id)
+                            client_context.api_key = ApiKey(resp["id"], resp["api_key"])
```

Comment:
```suggestion
                            client_context.api_key = ApiKey(id=resp["id"], secret=resp["api_key"])
```

Reply:
Done in https://github.com/elastic/rally/pull/1520/commits/d9d7ec5010ac540246360f5a053b9ae76c69a90e